core/build.gradle

```diff
@@ -0,0 +1,109 @@
+apply plugin: 'kotlin-multiplatform'
+
+task emptyJavadoc(type: Jar) {
+    classifier = 'javadoc'
+}
+
+kotlin {
+    infra {
+        shared {
+            compilations.main {
+                cinterops {
+                    bits { defFile = file("nativeMain/interop/bits.def") }
+                    sockets { defFile = file("nativeMain/interop/sockets.def") }
+                }
+            }
+            compilations.test {
+                cinterops {
+                    testSockets { defFile = file("nativeTest/interop/testSockets.def") }
+                }
+            }
+        }
+
+        target('macosX64')
+        target('iosX64')
+        target('iosArm64')
+        target('iosArm32')
+        target('linuxX64')
+        target('mingwX64')
```

Comment:
`kotlinx-serialization` also supports mingwX32 and linuxArm32Hfp targers (per community requests), so consider adding support for them. Also, Kotlin 1.3.60 will have new watchOS and tvOS targets.